.github/workflows/new-benchmark.yml

```diff
@@ -0,0 +1,56 @@
+name: New Benchmark
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
I think we might want to set up some ignore paths or something. This is going to be somewhat expensive (30mins for the validation benchmark, per commit), so maybe we should try and skip some commits that don't change anything?